src/TensorNetwork.jl

```diff
@@ -437,6 +437,9 @@ end
 function Base.replace!(tn::AbstractTensorNetwork, pair::Pair{<:Tensor,<:Tensor})
     tn = TensorNetwork(tn)
     old_tensor, new_tensor = pair
+
+    old_tensor == new_tensor && return tn
```

Comment:
mmm shouldn't this be `===` to check for egality, not equality?
```suggestion
    old_tensor === new_tensor && return tn
```

Reply:
No! Since for example, now it failed when we did `replace!(tn, A => conj(A))`, when `A` is a `Tensor` with a parent `Array` that are `Real`.

Reply:
Exactly! In that case, `A === conj(A)`!

Reply:
`conj(A)` returns a new object with is equal to `A`. It would be like you say if it is `conj!` (not `conj`).

Reply:
You're right with this last point, but the bug in #225 is when it's replaced with the *same* tensor (i.e. `old_tensor === new_tensor`).

I try to avoid `==` if posible because `==` iterates over the whole arrays and checks for equality element-wise, while `===` just checks the `objectid`s and you're done. Thus it avoids the overhead of iterating over everything, which can be costly on large arrays (specially when we start using distributed arrays).

Reply:
> but the bug in #225 is when it's replaced with the _same_ tensor (i.e. `old_tensor === new_tensor`)

Yes, but I tried `replace!(tn, A => conj(A))` and we got also the same error. I think this is because if you do `conj(A)` you still get the same `objectid`, which for me this is strange:
```julia
julia> objectid(A)
0xc7798e05eba779fe

julia> objectid(conj(A))
0xc7798e05eba779fe
```

But whatever, okay! If that is the case I agree that we can have `===`, I will push that commit now.